include/swift/AST/TypeCheckRequests.h

```diff
@@ -2080,6 +2080,27 @@ class IsCallableNominalTypeRequest
   bool isCached() const { return true; }
 };
 
+/// Lookup the nominal type \c callAsFunction method declarations if it has
+/// any.
+class NominalTypeCallAsFunctionDeclarationsRequest
+    : public SimpleRequest<NominalTypeCallAsFunctionDeclarationsRequest,
+                           ArrayRef<FuncDecl *>(CanType, DeclContext *),
+                           RequestFlags::Cached> {
```

Comment:
Since getting callable methods has now been requestified, I think you can remove `ConstraintSystem::getDynamicCallableMethods`?

Reply:
Not sure, I think right now this only handles `callAsFunction` maybe this request would have to be adapted to get `dynamicCall` too

Reply:
It looks like `ApplicableFn` constraint already does a lookup for `Id_callAsFunction` inside if it detects that other side is a callable type, so we might not need this request after all?

Reply:
Yeah, seems like not